handlers.go

```diff
@@ -181,7 +181,7 @@ func handleUnits(add func(Change), services map[string]*charm.ApplicationSpec, a
 			}
 			// Generate the changes required in order to place this unit, and
 			// retrieve the identifier of the parent change.
-			parentId := unitParent(add, p, records, addedMachines, servicePlacedUnits, getSeries(application, defaultSeries))
+			parentId := unitParent(add, p, records, addedMachines, servicePlacedUnits, getSeries(application, defaultSeries), application)
```

Comment:
Here, and in all subsequent calls, we could just pass application.Constraints directly rather than the whole spec. [shrug]